src/openforms/js/components/admin/form_design/data/complete-form.js

```diff
@@ -247,6 +247,20 @@ const saveVariables = async (state, csrftoken) => {
       // in the backend
       if (variable.dataType === 'boolean' && typeof variable.initialValue != 'boolean')
         variable.initialValue = variable.initialValue === 'true';
+
+      if (variable.serviceFetchConfiguration) {
+        if (Array.isArray(variable.serviceFetchConfiguration.headers)) {
+          variable.serviceFetchConfiguration.headers = Object.fromEntries(
+            variable.serviceFetchConfiguration.headers || []
```

Comment:
```suggestion
            variable.serviceFetchConfiguration.headers
```
this code path is only entered when it's proven to be an array via the `Array.isArray` check, so it can't be a falsy value (can't be null, undefined, false or empty string)